src/fontra/views/applicationsettings/panel-shortcuts.js

```diff
@@ -0,0 +1,462 @@
+      })
+    );
+
+    containerButtons.appendChild(
+      html.input({
+        type: "button",
+        style: `justify-self: start;`,
+        value: "Export shortcuts",
+        onclick: (event) => this.exportShortCuts(),
+      })
+    );
+
+    containerButtons.appendChild(
+      html.input({
+        type: "button",
+        style: `justify-self: start;`,
+        value: "Import shortcuts",
+        onclick: (event) => this.importShortCuts(),
+      })
+    );
+
+    this.panelElement.appendChild(containerButtons);
+    const shortCutsGrouped = getShortCutsGrouped();
+    for (const topic of Object.keys(shortCutsGrouped).sort()) {
+      const container = html.div({ class: "fontra-ui-shortcuts-panel" }, []);
+      container.appendChild(
+        html.createDomElement("div", {
+          class: "fontra-ui-shortcuts-panel-header",
+          innerHTML: translate(topic.slice(5)),
+        })
+      );
+      for (const actionIdentifier of shortCutsGrouped[topic]) {
+        container.appendChild(new ShortCutElement(actionIdentifier));
+      }
+      this.panelElement.appendChild(container);
+    }
+  }
+
+  async resetToDefault() {
+    const result = await dialog(
+      "Reset to default",
+      "Are you sure you want to reset all shortcuts to their default settings?",
+      [
+        { title: translate("dialog.cancel"), isCancelButton: true },
+        { title: "Okay", isDefaultButton: true },
+      ]
+    );
+    if (!result) {
+      return;
+    }
+    for (const actionIdentifier of getActionIdentifiers()) {
+      setCustomShortCuts(actionIdentifier, null);
+    }
+    location.reload();
+  }
+
+  async exportShortCuts() {
+    // Only export custom shortcuts,
+    // because default shortcuts are already in the code.
+    const shortCutsDataCustom = {};
+    for (const actionIdentifier of getActionIdentifiers()) {
+      const actionInfo = getActionInfo(actionIdentifier);
+      const shortCuts = actionInfo.customShortCuts;
+      if (!shortCuts) {
+        continue;
+      }
+      shortCutsDataCustom[actionIdentifier] = shortCuts;
+    }
+    const data = JSON.stringify(shortCutsDataCustom, null, 2);
+    const blob = new Blob([data], { type: "application/json" });
+    const url = URL.createObjectURL(blob);
+    const a = document.createElement("a");
+    a.href = url;
+    a.download = "fontra-shortcuts.json";
+    a.click();
+    URL.revokeObjectURL(url);
+  }
+
+  async importShortCuts() {
+    const input = document.createElement("input");
+    input.type = "file";
+    input.accept = ".json";
+    input.onchange = async (event) => {
+      const file = event.target.files[0];
+      if (!file) {
+        return;
+      }
+      const reader = new FileReader();
+      reader.onload = async (event) => {
+        const data = JSON.parse(event.target.result);
+        for (const actionIdentifier in data) {
+          setCustomShortCuts(actionIdentifier, data[actionIdentifier]);
```

Comment:
Please add a check here to ensure `actionIdentifier` is an existing one. Else `setCustomShortCuts()` will fail.

Reply:
Didn't you say we won't check that because it's covered in `setCustomShortCuts` anyway (and good to know if there is something wrong):
`  assert(actionInfo, `unknown actionIdentifier: ${actionIdentifier}`);`
Please don't get me wrong, I am happy to change it. I only want to double check with you if I misunderstood something.

Reply:
No. In the current state import will fail completely if it uses a non-existent key. This would happen if we'd rename or delete an action. So, the assert in `setCustomShortCuts()` is right, but the _caller_ can (and should in this case) filter for existing action identifiers.

Reply:
Thanks. done.